.github/workflows/release-tag.yml

```diff
@@ -0,0 +1,28 @@
+name: Tag the new release
+on:  
+  push:
+    tags: ["v*"]
+
+jobs:
+  build:
+    permissions:
+      contents: write
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - run: |
+          echo "GITHUB_REF is $GITHUB_REF"
+
+          [[ "$GITHUB_REF" =~ ^refs/tags/v[0-9]+\.[0-9]+\.[0-9]+$ ]] || (echo "$GITHUB_REF does not seem like a valid tag. expected format refs/tags/v10.2.3" && exit 1)
+          MAJOR_TAG=$(echo -n $GITHUB_REF | sed -e "s/^refs\/tags\/\(v[0-9]\+\)\.[0-9]\+.[0-9]\+$/\1/g")
+
+          [[ "$MAJOR_TAG" =~ ^v[0-9]+$ ]] || (echo "$MAJOR_TAG does not seem like a valid major tag" && exit 1)
+
+          echo "deleting existing tag $MAJOR_TAG"
+          git push origin :tags/$MAJOR_TAG || true
```

Comment:
Out of curiosity, I'm wondering if you have experience or opinions on the difference between this approach of deleting the tag and then recreating or just going directly to force-recreation via `git tag -f $MAJOR_TAG && git push -f origin tags/$MAJOR_TAG`?

Also, is there a difference when pushing between `git push origin $MAJOR_TAG` VS `git push origin tags/$MAJOR_TAG`?  I suppose the latter is useful if there may be a branch named $MAJOR_TAG?

Reply:
>Out of curiosity, I'm wondering if you have experience or opinions on the difference between this approach of deleting the tag and then recreating or just going directly to force-recreation via git tag -f $MAJOR_TAG && git push -f origin tags/$MAJOR_TAG?

No, I don't have opinion about one or the other. I like the one you mentioned better (haven't played around with tags a lot except `tag and forget`). 

>Also, is there a difference when pushing between git push origin $MAJOR_TAG VS git push origin tags/$MAJOR_TAG? I suppose the latter is useful if there may be a branch named $MAJOR_TAG?

yeah that is what I read (not in the official documentation though), and the recommendation was to be specific to use tags to be on safer side.